File: frontend/quiz/src/seleniumTests/Pages/ModalSignupPage.js
import Page from "./Page";
import { getElementById } from "./Utility";

const visibilityId = "closeModal";
const username = "usernameS";
const firstPassword = "passwordS";
const secondPassword = "password2";
const signupBtnId = "SingUp";

class ModalSignupPage extends Page {
  async isVisible() {
    return await super.isVisibleById(visibilityId);
  }

  get usernameBtn() {
    return getElementById(this.driver, username);
  }
  get passwordBtn() {
    return getElementById(this.driver, firstPassword);
  }
  get password2Btn() {
    return getElementById(this.driver, secondPassword);
  }
  get signUpBtn() {
    return getElementById(this.driver, signupBtnId);
  }

  async open(path) {
    await super.open(path);
  }
}

export default ModalSignupPage;

File: backend/selenium/ModalSignUpPage.py
from Page import Page
from Utility import getElementById, isVisibleById

visibilityId = "closeModal"
username = "usernameS"
firstPassword = "passwordS"
secondPassword = "password2"
signUpBtnId = "SingUp"


class ModalSignUpPage(Page):
    def isVisible(self):
        return isVisibleById(self.driver, visibilityId)

    def usernameBtn(self):
        return getElementById(self.driver, username)

    def passwordBtn(self):
        return getElementById(self.driver, firstPassword)

    def password2Btn(self):
        return getElementById(self.driver, secondPassword)

    def signUpBtn(self):
        return getElementById(self.driver, signUpBtnId)
